### ISSUE TITLE
Running in Docker error

### ISSUE THREAD

**Issue by Froghut** [haicenhacks/homeInventory#15]
I set everything up according to the readme, when I try to start the container I get the following error message:

```
Traceback (most recent call last):
  File "/home/app/web/manage.py", line 21, in <module>
    main()
  File "/home/app/web/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/usr/local/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.10/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/usr/local/lib/python3.10/site-packages/django/contrib/admin/apps.py", line 27, in ready
    self.module.autodiscover()
  File "/usr/local/lib/python3.10/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/usr/local/lib/python3.10/site-packages/django/utils/module_loading.py", line 47, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/local/lib/python3.10/site-packages/django/contrib/auth/admin.py", line 6, in <module>
    from django.contrib.auth.forms import (
  File "/usr/local/lib/python3.10/site-packages/django/contrib/auth/forms.py", line 11, in <module>
    from django.contrib.auth.tokens import default_token_generator
  File "/usr/local/lib/python3.10/site-packages/django/contrib/auth/tokens.py", line 117, in <module>
    default_token_generator = PasswordResetTokenGenerator()
  File "/usr/local/lib/python3.10/site-packages/django/contrib/auth/tokens.py", line 18, in __init__
    self.secret = self.secret or settings.SECRET_KEY
  File "/usr/local/lib/python3.10/site-packages/django/conf/__init__.py", line 90, in __getattr__
    raise ImproperlyConfigured("The SECRET_KEY setting must not be empty.")
django.core.exceptions.ImproperlyConfigured: The SECRET_KEY setting must not be empty.
```

I checked in docker and the DOCKER environment variable is set to TRUE, I also tried explicitly setting a SECRET_KEY environment variable but I get the same error.

**Comment by haicenhacks** [comment]
How are you setting the secret key variable?

the project.env file should look like this:
```
POSTGRES_USER=django
POSTGRES_PASSWORD=password
POSTGRES_DB=project_db

DATABASE=postgres
DATABASE_HOST=postgresdb_homeInventory
DATABASE_PORT=5432
DJANGO_SECRET_KEY=testing123
DOCKER=TRUE
DJANGO_SETTINGS_MODULE=home.settings.testing
```

**Comment by Froghut** [comment]
That did the trick, using DJANGO_SECRET_KEY instead of SECRET_KEY.  Thank you!
Although having a quick look at the code I'm not sure why it did not use the "hardcoded" secret key because of the DOCKER=TRUE env variable?

the container now starts correctly and I can access the website. I tried creating the superuser using the command 
`django-admin createsuperuser --no-input --username=admin --email admin@example.com --password pass`
in the cli of the docker container, but I get the following error:
```
No Django settings specified.
Unknown command: 'createsuperuser'
Type 'django-admin help' for usage.
```

**Comment by haicenhacks** [comment]
I think everything is currently broken. I tried doing some manual testing, but it seems like a problem with postgresql not applying migrations correctly. Won't have any time to troubleshoot this for the next several weeks. I don't know how comfortable you are with making modifications, but swapping out postgres for sqlite as the back end might make things work temporarily.

**Comment by Froghut** [comment]
Edit: managed to get it working!
I needed to set two more environment variables:
```
DJANGO_SETTINGS_MODULE=home.settings.production
PYTHONPATH=/home/app/web
```
I also changed the Database backend to SQLite in home/settings/base.py - I changed the db file path and mounted that folder in docker. I also had to create the db.sqlite3 file manually - an empty file worked.

after that I was able to create a superuser in the docker cli using `django-admin createsuperuser --username=admin --email admin@example.com`

**Comment by haicenhacks** [comment]
Thanks for the feedback. I should warn you that unless you modified the docker-compose file to add a mount for the database directory that your changes won't persist across docker builds.

I'm working on getting that pushed to the repository.

**Comment by haicenhacks** [comment]
Everything should be fixed as of [aa3a76b](https://github.com/haicenhacks/homeInventory/commit/aa3a76b1ae51f534f648010dd16d86d968c5ca45)
